Lib/functools.py

```diff
@@ -273,43 +273,123 @@ def reduce(function, sequence, initial=_initial_missing):
 ### partial() argument application
 ################################################################################
 
+
+class __PlaceholderTypeBase:
+    """The type of the Placeholder singleton.
+
+    Used as a placeholder for partial arguments.
+    """
+    __instance = None
+
+    def __new__(cls):
+        if cls.__instance is None:
+            cls.__instance = object.__new__(cls)
+        return cls.__instance
+
+    def __bool__(self):
+        raise TypeError("Placeholder should not be used in a boolean context")
+
+    def __repr__(self):
+        return 'Placeholder'
+
+    def __reduce__(self):
+        return 'Placeholder'
```

Comment:
```suggestion
    def __repr__(self):
        return '<Placeholder>'

    def __init_subclass__(cls, *args, **kwargs):
        raise TypeError(f"type '{cls.__name__}' is not an acceptable base type")
```

Should prohibit subclassing to align with C implementation.

```python
In [1]: import functools

In [2]: class Foo(type(functools.Placeholder)):
   ...:     pass
   ...:     
Unexpected exception formatting exception. Falling back to standard exception
Traceback (most recent call last):
  File "/Users/PanXuehai/Projects/cpython/venv/lib/python3.14/site-packages/IPython/core/interactiveshell.py", line 3577, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
    ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<ipython-input-2-cf7bdbc02cef>", line 1, in <module>
    class Foo(type(functools.Placeholder)):
        pass
TypeError: type 'functools.PlaceholderType' is not an acceptable base type
```